### ISSUE TITLE
Meson sandbox violation for xdg-shell.xml

### ISSUE THREAD

**Issue by stevenengler** [swaywm/sway#6878]
When building sway (HEAD = 04676936e71f6f) on Ubuntu 20.04 with the latest meson (0.61.2), the build fails due to the error:

```
protocols/meson.build:37:1: ERROR: Sandbox violation: Tried to grab file xdg-shell.xml from a nested subproject.
```

https://github.com/swaywm/sway/blob/04676936e71f6fccccb098f3232d16572b140902/protocols/meson.build#L37-L42

This can be fixed by changing `input: xml` to `input: files(xml)`. This is also an issue with the following two `custom_target()` functions.

**Comment by emersion** [comment]
Yeah this is a Meson bug, hopefully will be fixed by https://github.com/mesonbuild/meson/pull/10039.
